### PR TITLE
[nrfconnect] Bump nRF Connect SDK version in Docker

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
Regular update of nRF Connect SDK to 2.4.0 version

### CODE REVIEW

File: integrations/docker/images/chip-build-nrf-platform/Dockerfile [hunk 1]
```diff
@@ -6,7 +6,7 @@ ARG VERSION=latest
 
 FROM connectedhomeip/chip-build:${VERSION} as build
 # Compatible Nordic Connect SDK revision.
-ARG NCS_REVISION=v2.3.0
```
Comment (Damian-Nordic) [inline integrations/docker/images/chip-build-nrf-platform/Dockerfile:9]: Might be also a good idea to update zephyr-sdk to 0.16.0 and west to 1.0.0 below to match what our toolchain manager installs.

**Review by Damian-Nordic — APPROVED** [review]

File: integrations/docker/images/chip-build-nrf-platform/Dockerfile (hunk 1 above)
Comment (kkasperczyk-no) [reply integrations/docker/images/chip-build-nrf-platform/Dockerfile:9]: Good point, done.

**Review by andy31415 — APPROVED** [review]

File: scripts/build/builders/nrf.py [hunk 2]
```diff
@@ -161,7 +161,7 @@ def generate(self):
                 zephyr_base = os.environ['ZEPHYR_BASE']
                 nrfconnect_sdk = os.path.dirname(zephyr_base)
                 zephyr_sdk_dir = os.environ.get('ZEPHYR_SDK_INSTALL_DIR') or os.path.join(
-                    os.environ['NRF5_TOOLS_ROOT'], 'zephyr-sdk-0.15.2')
+                    os.environ['NRF5_TOOLS_ROOT'], 'zephyr-sdk-0.16.0')
```
Comment (andy31415) [inline scripts/build/builders/nrf.py:164]: in hindsight this was NOT safe: this makes things break for all builders that have not yet upgraded to the new dockerfile.

Comment (andy31415) [reply scripts/build/builders/nrf.py:164]: change reverted in #27015

**Pull request merged**